eZ/Bundle/EzPublishCoreBundle/Command/DebugServiceCommand.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishCoreBundle\Command;
+
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DebugServiceCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}.
+     */
+    public function configure()
+    {
+        $this->setName('ezplatform:debug:service');
+        $this->setDescription('Debug / Retrive class name of a service');
+        $this->addArgument(
+            'service',
+            InputArgument::REQUIRED,
+            'The service to return class name for, for instance "ezpublish.cache_pool"'
+        );
+        $this->addOption(
+            'oneline',
+            'o',
+            InputOption::VALUE_NONE,
+            'Only return value, for automation / testing use on a single line.'
+        );
+        $this->setHelp(<<<EOM
+Outputs a given service class name.
+
+To rather see service definition, use: <comment>debug:container ezpublish.cache_pool</comment>
+
+EOM
+        );
+    }
+
+    /**
+     * {@inheritdoc}.
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $service = $input->getArgument('service');
+        $className = get_class($this->getContainer()->get($service));
```

Comment:
debug:container shows afaik just definition, not the actual class on the instance.

But indeed, this will throw on symfony 4 for any private service.
And maybe this is more interesting to contribute to symfony instead, so we can drop this command if your against it.

Reply:
It shows everything what you need and more.
```
$ php ./bin/console debug:container ezpublish.siteaccess

Information for Service "ezpublish.siteaccess"
==============================================

 ---------------- ---------------------------------------- 
  Option           Value                                   
 ---------------- ---------------------------------------- 
  Service ID       ezpublish.siteaccess                    
  Class            eZ\Publish\Core\MVC\Symfony\SiteAccess  
  Tags             -                                       
  Public           no                                      
  Synthetic        no                                      
  Lazy             no                                      
  Shared           yes                                     
  Abstract         no                                      
  Autowired        no                                      
  Autoconfigured   no                                      
 ---------------- ----------------------------------------
```
```
Options:
      --show-private             Used to show public *and* private services
      --show-arguments           Used to show arguments in services
      --tag=TAG                  Shows all services with a specific tag
      --tags                     Displays tagged services for an application
      --parameter=PARAMETER      Displays a specific parameter for an application
      --parameters               Displays parameters for an application
      --types                    Displays types (classes/interfaces) available in the container
      --format=FORMAT            The output format (txt, xml, json, or md) [default: "txt"]
```
Honestly, it's reinventing the wheel :P

Reply:
If so, and if that is the case on earlier symfony versions too then +1
It's dropped.

Reply:
@alongosz Sorry, re-opening thread here as I realize I didn't manage to explain to you what this was meant to do.

The Class in output from `debug:container` is as I tried to say above, just outputs what is defined in `yml`, while the command here was about giving you the *actual* class.

As most of those are proxies, if we add this to it:
```
        if ($service instanceof VirtualProxyInterface) {
            $service->initializeProxy();
            $output->writeln("<comment>Wrapped class:</comment> ". get_class($service->getWrappedValueHolderValue()));
        }
```

To show the difference for for instance `ezpublish.image_alias.imagine.cache_resolver_decorator`:

```
Class                    Liip\ImagineBundle\Imagine\Cache\Resolver\ProxyResolver 
```
vs:
```
Wrapped class:   eZ\Bundle\EzPublishCoreBundle\Imagine\IORepositoryResolver
```

So the point of this is to be able to debug the result of factories.
E.g. to debug that you really managed to configure Varnish correctly with ezplatform-http-cache for instance, and lots of similar cases.

So it's not the same, but we can leave it out of this PR of you still think so. But I DO think this provides value, so then please give some feedback on in which form you think it rather should be in.

Reply:
To sum up this thread talked to @alongosz on slack and this part we can try to contribute to symfony or something, if they have a way to introspect private services too.